# HG changeset patch
# User bpb
# Date 1489497830 0
#      Tue Mar 14 13:23:50 2017 +0000
# Node ID 51631f9fa8d8b9b23b8dc129285f3264b5b7a1f7
# Parent  d7bd49ad8f0a6dd3e9865cacd4354a2916137333
8172465: Better handling of channel groups
Reviewed-by: alanb, coffeys, skoivu, rhalade, chegar

--- jdk/src/share/classes/sun/nio/ch/AsynchronousChannelGroupImpl.java
+++ jdk/src/share/classes/sun/nio/ch/AsynchronousChannelGroupImpl.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2008, 2009, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2008, 2017, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -246,13 +246,16 @@ abstract class AsynchronousChannelGroupI
     abstract void shutdownHandlerTasks();
 
     private void shutdownExecutors() {
-        AccessController.doPrivileged(new PrivilegedAction<Void>() {
-            public Void run() {
-                pool.executor().shutdown();
-                timeoutExecutor.shutdown();
-                return null;
-            }
-        });
+        AccessController.doPrivileged(
+            new PrivilegedAction<Void>() {
+                public Void run() {
+                    pool.executor().shutdown();
+                    timeoutExecutor.shutdown();
+                    return null;
+                }
+            },
+            null,
+            new RuntimePermission("modifyThread"));
     }
 
     @Override
